### PR TITLE
QElecProdCos added

### PR DESCRIPTION
**Pull request by riazis**
-new equation added

### CODE REVIEW

File: declarations.gms [hunk 1]
```diff
@@ -1,13 +1,20 @@
 Parameters
-iCGI(allCy,YTIME)         "Capital Goods Index (defined as CGI(Scenario)/CGI(Baseline))"
-iNPDL(SBS)                "Number of Polynomial Distribution Lags (PDL)"
-iFPDL(SBS,KPDL)           "Polynomial Distribution Lags (PDL) Coefficients per subsector"
+iCGI(allCy,YTIME)                           "Capital Goods Index (defined as CGI(Scenario)/CGI(Baseline))"
+iNPDL(SBS)                                  "Number of Polynomial Distribution Lags (PDL)"
+iFPDL(SBS,KPDL)                             "Polynomial Distribution Lags (PDL) Coefficients per subsector"
+iResDemSub(allCy,SBS,YTIME)                 "Residuals in total energy demand per subsector (energy intensity)"
```
Comment (giannou) [inline declarations.gms:5]: "energy intensity" is not a unit

File: equations.gms [hunk 2]
```diff
@@ -136,9 +136,32 @@ QElecIndPricesEst(runCy,YTIME)$TIME(YTIME)..
...
 
+* Compute electricity production cost per CHP plant and demand sector in Euro per KWh
+QElecProdCos(runCy,DSBS,CHP,YTIME)$(TIME(YTIME) $INDDOM(DSBS))..
+         VElecProdCost(runCy,DSBS,CHP,YTIME)
+                 =E=
+                  ( (
+                    ( ( iDisc(runCy,"PG",YTIME) * exp(iDisc(runCy,"PG",YTIME)*iLifChpPla(CHP))
+                        / (exp(iDisc(runCy,"PG",YTIME)*iLifChpPla(CHP)) -1))
+                      * iCapCosChp(runCy,CHP,YTIME)* 1E3 * iCGI(runCy,YTIME)  + iFixOandMCosChp(runCy,CHP,YTIME)
+                    )/(iAvailRateChp(runCy,CHP)*8.76)/1000
```
Comment (giannou) [inline equations.gms:154]: please define a scalar (or use an existing) for 8.76

File: equations.gms [hunk 3]
```diff
@@ -136,9 +136,32 @@ QElecIndPricesEst(runCy,YTIME)$TIME(YTIME)..
...
+*( ((VFuelPriceSub(runCy,DSBS,EF,YTIME) + (VRenValue(YTIME)/1000)$(not RENEF(EF))+VCosTech(runCy,DSBS,EF,TEA,YTIME)/1000)/iUseEneConvSubTech(DSBS,EF,YTIME)- (0$(not CHP(EF)) + (VFuelPriceSub(runCy,"OI","ELC",YTIME)*iFracElecPriChp(runCy,YTIME)*VElecIndPrices(runCy,YTIME))$CHP(EF))) + (0.003) + SQRT( SQR(((VFuelPriceSub(runCy,DSBS,EF,YTIME)+VCosTech(runCy,DSBS,EF,TEA,YTIME)/1000)/VCosTech(runCy,DSBS,EF,TEA,YTIME)- (0$(not CHP(EF)) + (VFuelPriceSub(runCy,"OI","ELC",YTIME)*iFracElecPriChp(runCy,YTIME)*VElecIndPrices(runCy,YTIME))$CHP(EF)))-(0.003)) + SQR(1e-7) ) )/2;
 
+* Compute electricity production cost per CHP plant and demand sector in Euro per KWh
+QElecProdCos(runCy,DSBS,CHP,YTIME)$(TIME(YTIME) $INDDOM(DSBS))..
+         VElecProdCost(runCy,DSBS,CHP,YTIME)
+                 =E=
+                  ( (
+                    ( ( iDisc(runCy,"PG",YTIME) * exp(iDisc(runCy,"PG",YTIME)*iLifChpPla(CHP))
+                        / (exp(iDisc(runCy,"PG",YTIME)*iLifChpPla(CHP)) -1))
+                      * iCapCosChp(runCy,CHP,YTIME)* 1E3 * iCGI(runCy,YTIME)  + iFixOandMCosChp(runCy,CHP,YTIME)
```
Comment (giannou) [inline equations.gms:153]: please change `1E3` to `1000`

File: equations.gms [hunk 4]
```diff
@@ -136,9 +136,32 @@ QElecIndPricesEst(runCy,YTIME)$TIME(YTIME)..
...
+                    )/(iAvailRateChp(runCy,CHP)*8.76)/1000
+                    + iCosPerChp(runCy,CHP,YTIME)/1E3
+
+
```
Comment (giannou) [inline equations.gms:157]: please remove empty lines

File: equations.gms [hunk 5]
```diff
@@ -136,9 +136,32 @@ QElecIndPricesEst(runCy,YTIME)$TIME(YTIME)..
...
+                    + sum(PGEF$CHPtoEF(CHP,PGEF), (VFuelPriceSub(runCy,"PG",PGEF,YTIME)+1e-3*iCo2EmiFac(runCy,"PG",PGEF,YTIME)*
+                         (sum(NAP$NAPtoALLSBS(NAP,"PG"),VCarVal(runCy,NAP,YTIME))))
+                         *0.086/(   iBoiEffChp(runCy,CHP,YTIME)*VElecIndPrices(runCy,YTIME)    )
```
Comment (giannou) [inline equations.gms:160]: is `0.086` the scalar `sTWhToMtoe` maybe?

**Review by giannou — CHANGES_REQUESTED** [review]

File: equations.gms [hunk 6]
```diff
@@ -136,9 +136,32 @@ QElecIndPricesEst(runCy,YTIME)$TIME(YTIME)..
...
-        (VFuelPriceSub(runCy,"OI","ELC",YTIME-3)/VFuelPrice(runCy,"OI",YTIME-3)))**(0.3); 
+        (VFuelPriceSub(runCy,"OI","ELC",YTIME-3)/VFuelPrice(runCy,"OI",YTIME-3)))**(0.3);
 
+* Compute fuel prices per subsector and fuel especially for chp plants (take into account the profit of electricity sales)
+*QFuePriSubChp(runCy,DSBS,EF,TEA,YTIME)$(TIME(YTIME) $(not TRANSE(DSBS))  $SECTTECH(DSBS,EF) )..
+*VFuePriSubChp(runCy,DSBS,EF,TEA,YTIME) =E=   
+*( ((VFuelPriceSub(runCy,DSBS,EF,YTIME) + (VRenValue(YTIME)/1000)$(not RENEF(EF))+VCosTech(runCy,DSBS,EF,TEA,YTIME)/1000)/iUseEneConvSubTech(DSBS,EF,YTIME)- (0$(not CHP(EF)) + (VFuelPriceSub(runCy,"OI","ELC",YTIME)*iFracElecPriChp(runCy,YTIME)*VElecIndPrices(runCy,YTIME))$CHP(EF))) + (0.003) + SQRT( SQR(((VFuelPriceSub(runCy,DSBS,EF,YTIME)+VCosTech(runCy,DSBS,EF,TEA,YTIME)/1000)/VCosTech(runCy,DSBS,EF,TEA,YTIME)- (0$(not CHP(EF)) + (VFuelPriceSub(runCy,"OI","ELC",YTIME)*iFracElecPriChp(runCy,YTIME)*VElecIndPrices(runCy,YTIME))$CHP(EF)))-(0.003)) + SQR(1e-7) ) )/2;
 
+* Compute electricity production cost per CHP plant and demand sector in Euro per KWh
+QElecProdCos(runCy,DSBS,CHP,YTIME)$(TIME(YTIME) $INDDOM(DSBS))..
```
Comment (giannou) [inline equations.gms:147]: An execution error (division by zero) happens somewhere in this equation.

Comment (giannou) [inline equations.gms:147]: An execution error (division by zero) happens somewhere in this equation.

**Review by giannou — CHANGES_REQUESTED** [review]

File: declarations.gms (hunk 1 above)
Comment (riazis) [reply declarations.gms]: fixed

File: equations.gms (hunk 3 above)
Comment (riazis) [reply equations.gms]: fixed

File: equations.gms (hunk 4 above)
Comment (riazis) [reply equations.gms]: fixed

File: equations.gms (hunk 2 above)
Comment (giannou) [reply equations.gms]: fixed

File: equations.gms (hunk 5 above)
Comment (giannou) [reply equations.gms]: fixed

File: equations.gms (hunk 6 above)
Comment (giannou) [reply equations.gms]: fixed

**Review by giannou — APPROVED** [review]
great, thanks

**Pull request merged**